turbopack/crates/turbopack-ecmascript/src/references/external_module.rs

```diff
@@ -82,19 +85,44 @@ impl CachedExternalModule {
     pub fn content(&self) -> Result<Vc<EcmascriptModuleContent>> {
         let mut code = RopeBuilder::default();
 
-        if self.external_type == CachedExternalType::EcmaScriptViaImport {
-            writeln!(
-                code,
-                "const mod = await {TURBOPACK_EXTERNAL_IMPORT}({});",
-                StringifyJs(&self.request)
-            )?;
-        } else {
-            writeln!(
-                code,
-                "const mod = {TURBOPACK_EXTERNAL_REQUIRE}({}, () => require({}));",
-                StringifyJs(&self.request),
-                StringifyJs(&self.request)
-            )?;
+        match self.external_type {
+            CachedExternalType::EcmaScriptViaImport => {
+                writeln!(
+                    code,
+                    "const mod = await {TURBOPACK_EXTERNAL_IMPORT}({});",
+                    StringifyJs(&self.request)
+                )?;
+            }
+            CachedExternalType::Global => {
+                if self.request.is_empty() {
+                    writeln!(code, "const mod = {{}};")?;
+                } else if self.request.contains('/') {
```

Comment:
(asking because I did a quick search and couldn't easily find an answer) How does webpack do this? Does it use `'/'` or `'.'`, or does it just not support nested objects in this string?

We could just make you import/require the top-level object, and then it's up to the import to do the property access. This is an external, so concerns like fine-grained dependency tracking and tree-shaking don't apply.

Reply:
this seems invalid here, webpack can deal the case like: 

```js
externals: {
  'antd/es/button': ['global', 'antd', 'es', 'button']
}
```

it can generate `global['antd']['es']['button']`, this is becase the `request` of webpack can accept both `string` and `string[]`: https://github.com/webpack/webpack/blob/main/lib/ExternalModule.js#L727

For turbopack case, the request now only accept a `RcStr`:

https://github.com/vercel/next.js/blob/canary/turbopack/crates/turbopack-ecmascript/src/references/external_module.rs#L61

I think i can move this branch here, and later i will send a pr to turbopack for the `external.request` to accept both `RcStr` and `Vec<RcStr>`, it will align with webpack